plugins/optimization-detective/optimization.php

```diff
@@ -73,14 +73,18 @@ function od_maybe_add_template_output_buffer_filter() {
  * @return bool Whether response can be optimized.
  */
 function od_can_optimize_response(): bool {
+
+	// Retrieve the sanitized value of the 'REQUEST_METHOD' server variable.
+	$method = filter_input( INPUT_SERVER, 'REQUEST_METHOD', FILTER_SANITIZE_STRING );
```

Comment:
I don't think we should use `filter_input()` for two reasons:

1. It's not used in core.
2. It's not able to be unit tested (e.g. by setting `$_SERVER['REQUEST_METHOD']`).

You can see there is a unit test failing for the second reason: https://github.com/WordPress/performance/actions/runs/8651242612/job/23721516648?pr=1123#step:9:84

You can see the test setup is overriding `$_SERVER['POST_METHOD']`: 

https://github.com/WordPress/performance/blob/2fd48b05409136580dd6bbdd4812dde95ccdd389/tests/plugins/optimization-detective/optimization-tests.php#L124-L130

Reply:
Got it, I have Updated this

Thanks @westonruter